liver_imaging_analysis/engine/engine.py

```diff
@@ -379,8 +397,9 @@ def fit(
                 self.optimizer.step()
                 training_loss += loss.item()
                 if batch_callback_epochs is not None:
-                    if (epoch + 1) % batch_callback_epochs == 0:
+                    if (epoch + 1) % batch_callback_epochs == 0 and summary_writer!= None:
```

Comment:
If no summary writer is provided, does it make sense not to call the per-batch callback? Also, avoid comparing None with !=, use "is not None" instead.

Reply:
This specific callback only writes in summary writer, it doesn't print or do anything else. Should I still move the condition inside the function?

Reply:
You can leave it as it is for now, but it doesn't look very clean. What if we add other functions in the callbacks that are not related to the summary writer? The engine should be as independent as possible from the models implementation.